components/script_bindings/domstring.rs

```diff
@@ -336,19 +357,23 @@ impl DOMString {
         StringView(self.0.borrow())
     }
 
+    /// Use this if you want to work on the `EncodedBytes` directly.
+    /// This will not do any conversions for you.
+    pub fn view(&self) -> EncodedBytesView<'_> {
+        EncodedBytesView(self.0.borrow())
+    }
+
     pub fn clear(&mut self) {
         *self.0.borrow_mut() = DOMStringType::Rust(String::new())
     }
 
     pub fn is_empty(&self) -> bool {
-        self.make_rust();
-        self.str().is_empty()
+        self.view().is_empty()
     }
 
     /// This length (as rust spec) is in bytes not chars.
     pub fn len(&self) -> usize {
-        self.make_rust();
-        self.str().len()
+        self.view().len()
     }
```

Comment:
I think this is will yield unexpected results, because the length of the byte sequence depends on whether the string is encoded in latin1 or utf8.  For example, `é` is one byte in latin1 and 2 bytes in utf8. It might be okay to have the length of the string suddenly change? I'm not sure.

Reply:
The length does not change, no matter the representation. I will add a test that actually enforces this.
The length method is used in a couple of places. Webdriver, Textinput, some elements, and bluetooth.